src/App.tsx

```diff
@@ -14,12 +14,55 @@ import Login from "./components/login/login";
 import Callback from "./components/login/callback";
 import Register from "./components/register/register";
 import Profile from "./components/login/profile";
+import authService from "./services/auth";
+import { useLayoutEffect } from "react";
 
-function App() {
-  return (
-    <BrowserRouter>
-      <Header />
-      <Routes>
+
+function RouterWrapper() {
+  
+  let location = useLocation();
+  const navigate = useNavigate()
+
+  useLayoutEffect(() => {
+    authService.getUser().then((user) => {
+      if ((!user?.id || user.changed) && location.pathname !== "/register" && user !== null) {
```

Comment:
```suggestion
      if (user && (!user.id || user.changed) && location.pathname !== "/register") {
```

First check whether user exists before checking user properties.